tests/chainer_tests/functions_tests/array_tests/test_get_item.py

```diff
@@ -10,179 +10,148 @@
-     {'slices': (numpy.array(True)),
-         'input_shape': (), 'sliced_shape': (1,)},
+    [{'slices': []},
+     {'slices': ([],)},
+     {'slices': ([[]],)},
+     {'slices': numpy.array([], dtype=numpy.bool)},
+     {'slices': (1, [1])},
+     {'slices': ([1], slice(1, 2))},
+     {'slices': [1, 0]},
+     {'slices': ([1, 0],)},
+     {'slices': numpy.array([[1, 0], [2, 3]])},
+     {'slices': ([1, 0], [1, 1])},
+     {'slices': ([1, 0], slice(None), [[1, 1], [1, 1]])},
+     {'slices': ([1, 0], slice(1, 2), [0, 0])},
+     {'slices': ([[1, 1], [1, 0]], slice(1, 2), 1)},
+     {'slices': numpy.array([True] * 18 + [False] * 6).reshape(4, 3, 2)},
+     {'slices': numpy.array([True, False, False, True])},
+     {'slices': (slice(None), numpy.array([True, False, True]))},
+     {'slices': numpy.array([False, False, False, False])},
+     {'slices': (3, 2, Ellipsis, 1)},
+     {'slices': (numpy.array(False))},
+     {'slices': (numpy.array(True))},
      ]
 ))
-class TestGetItemAdvanced(unittest.TestCase):
+class TestGetItemAdvanced(testing.FunctionTestCase):
 
     input_shape = (4, 3, 2)
 
     def setUp(self):
-        self.x_data = numpy.random.uniform(
-            -1, 1, self.input_shape).astype(self.dtype)
-        self.gy_data = numpy.random.uniform(
-            -1, 1, self.sliced_shape).astype(self.dtype)
-
-        self.check_backward_options = {'atol': 5e-4, 'rtol': 5e-4}
-
-    def check_forward(self, x_data):
-        x = chainer.Variable(x_data)
-        y = functions.get_item(x, self.slices)
-        self.assertEqual(y.data.dtype, self.dtype)
-        numpy.testing.assert_equal(cuda.to_cpu(x_data)[self.slices],
-                                   cuda.to_cpu(y.data))
-
-    def test_forward_cpu(self):
-        self.check_forward(self.x_data)
-
-    @attr.gpu
-    def test_forward_gpu(self):
-        self.check_forward(cuda.to_gpu(self.x_data))
-
-    def check_backward(self, x_data, y_grad):
-        def f(x):
-            return functions.get_item(x, self.slices)
-
-        gradient_check.check_backward(
-            f, (x_data,), y_grad, dtype='d', **self.check_backward_options)
-
-    def test_backward_cpu(self):
-        self.check_backward(self.x_data, self.gy_data)
+        self.check_backward_options.update({'atol': 5e-4, 'rtol': 5e-4})
+        self.check_double_backward_options.update({'atol': 1e-3, 'rtol': 1e-3})
+
+    def generate_inputs(self):
+        x = numpy.random.uniform(-1, 1, self.input_shape).astype(self.dtype)
+        return x,
+
+    def _convert_slices(self, slices, device):
+        # Converts advanced indexing slices (of numpy.ndarray) to respective
+        # backend arrays.
+        if isinstance(slices, list):
+            return [self._convert_slices(a, device) for a in slices]
+        if isinstance(slices, tuple):
+            return tuple([self._convert_slices(a, device) for a in slices])
+        if isinstance(slices, numpy.ndarray):
+            return device.send(slices)
```

Comment:
This PR changes the test to always use slices with the same "xp" when `slices` is an ndarray. It used to always use slices of `np.ndarray`. Both seems to be supported.